x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticator.java

```diff
@@ -135,6 +138,15 @@
  */
 public class OpenIdConnectAuthenticator {
 
+    static {
+        JSONValue.registerWriter(Nonce.class, new JsonWriterI<Nonce>() {
```

Comment:
json-smart uses reflection for serializing classes, unless they have a registered writer. This is turn would require createClassLoader permissions. 

`Nonce` is the only class where this is potentially an case (and the issue only surfaces in unit test code where we manually construct claim sets involving nonces as claims). Registering a write here lets us avoid the policy grants.

Reply:
This needs a better spot though...